### PR TITLE
add gh actions for building

### PR DESCRIPTION
**Pull request by xmnlab**
this PR adds an initial Github Actions for package building for macOS.

### CODE REVIEW

**Comment by xmnlab** [comment]
not sure why the GH-Actions was not executed here (maybe because it was not merged it) ... I tried that on my fork and the build worked (https://github.com/xmnlab/omnisci-datascience-installer/runs/1131689172?check_suite_focus=true).

is it ok to merge this PR before I implement the upload to github packages?

File: .github/workflows/package.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      matrix:
+        python-version: [3.7, 3.8]
+
+    env:
+      PYTHON_VERSION: ${{ matrix.python-version }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: goanpeca/setup-miniconda@v1
```
Comment (scopatz) [inline .github/workflows/package.yaml:23]: I think there is a miniforge one too

Comment (scopatz) [reply .github/workflows/package.yaml:23]: CC @goanpeca

File: .github/workflows/package.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  macos:
+
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8]
```
Comment (scopatz) [inline .github/workflows/package.yaml:15]: We probably just need one python version.

File: .github/workflows/package.yaml (hunk 1 above)
Comment (goanpeca) [reply .github/workflows/package.yaml:23]: >I think there is a miniforge one too

Not yet 🙃 but soon.

Also @xmnlab, please use

`conda-incubator/setup-miniconda@v1`

**Comment by goanpeca** [comment]
>not sure why the GH-Actions was not executed here (maybe because it was not merged it) .

Yes  @xmnlab, the first time you create a workflow it will not run for security reasons, you could do "evil" things from a fork.

When I do PRs that add GH actions, I usually push to the repo directly (not to the fork) so I can run and debug the things directly.

File: .github/workflows/package.yaml (hunk 1 above)
Comment (xmnlab) [reply .github/workflows/package.yaml:23]: thanks @scopatz and @goanpeca !

**Comment by xmnlab** [comment]
@goanpeca and @scopatz I already applied the suggestion you made. thanks for the feedback.

File: .github/workflows/package.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    - uses: conda-incubator/setup-miniconda@v1
+      with:
+        auto-update-conda: true
+        python-version: ${{ matrix.python-version }}
+        activate-environment: omnisci-ds-installer
+        channels: conda-forge
+
+    - name: install dependencies
+      shell: bash -l {0}
+      run: conda install constructor
```
Comment (scopatz) [inline .github/workflows/package.yaml:32]: I think you might need to grab the conda-standalone with wget, as stated in the readme

Comment (xmnlab) [reply .github/workflows/package.yaml:32]: oh I see .. sorry .. I miss understood this part, I thought it was required just for windows. I am fixing that now. thanks @scopatz !

**Comment by xmnlab** [comment]
@scopatz I added the suggestion you made. let me know if I am missing anything. thank you so much!

**Comment by scopatz** [comment]
Thanks @xmnlab!

**Pull request merged**